cpp/common/src/codingstandards/cpp/EncapsulatingFunctions.qll

```diff
@@ -18,6 +18,39 @@ class MainFunction extends MainLikeFunction {
   }
 }
 
+/**
+ * A test function from the GoogleTest infrastructure.
+ *
+ * Such functions can be treated as valid EntryPoint functions during analysis
+ * of "called" or "unused" functions. It is not straightforward to identify
+ * such functions, however, they have certain features that can be used for
+ * identification. This can be refined based on experiments/real-world use.
+ */
+class GTestFunction extends MainLikeFunction {
+  GTestFunction() {
+    // A GoogleTest function is named "TestBody" and
+    this.hasName("TestBody") and
+    // is enclosed by a class that inherits from a base class
+    this.getEnclosingAccessHolder() instanceof Class and
+    exists(Class base |
+      base = this.getEnclosingAccessHolder().(Class).getABaseClass() and
+      (
+        // called "Test" or
+        exists(Class c | base.getABaseClass() = c and c.hasName("Test"))
+        or
+        // defined under a namespace called "testing" or
+        exists(Namespace n | n = base.getNamespace() | n.hasName("testing"))
+        or
+        // is templatized by a parameter called "gtest_TypeParam_"
+        exists(TemplateParameter tp |
+          tp = base.getATemplateArgument() and
+          tp.hasName("gtest_TypeParam_")
+        )
+      )
+    )
+  }
+}
```

Comment:
We would also need to add a test case for Google Test entry points. This would require adding a stub for Google Test to `cpp/common/test/includes/googletest/gtest.h`, update the [options](https://github.com/github/codeql-coding-standards/blob/main/cpp/common/test/options) file to add the new google test directory to the include path, then referencing the relevant stubs in your test.

Reply:
I understand. I will work on this.

Reply:
Addressed in [c5cdf0e](https://github.com/github/codeql-coding-standards/pull/725/commits/c5cdf0e0b7e40239966e5bfb0127264f0b1e5627)